pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -480,15 +481,14 @@ func (b *volumeBinder) AssumePodVolumes(logger klog.Logger, assumedPod *v1.Pod,
 	// Assume PVCs
 	newProvisionedPVCs := []*v1.PersistentVolumeClaim{}
 	for _, claim := range podVolumes.DynamicProvisions {
-		// The claims from method args can be pointing to watcher cache. We must not
-		// modify these, therefore create a copy.
-		claimClone := claim.DeepCopy()
-		metav1.SetMetaDataAnnotation(&claimClone.ObjectMeta, volume.AnnSelectedNode, nodeName)
-		err = b.pvcCache.Assume(claimClone)
-		if err != nil {
-			b.revertAssumedPVs(newBindings)
-			b.revertAssumedPVCs(newProvisionedPVCs)
-			return
+		claimClone, dirty := annotateClaimWithNodeName(claim, nodeName)
```

Comment:
I would suggest to always copy the claim her to avoid potential modification in the downstream. And if you insist on changing this piece of code, let's separate it to another PR and focus on retry logic here.

Reply:
yes, I will fix this.